ee/src/main/java/org/jboss/as/ee/structure/InitializeInOrderProcessor.java

```diff
@@ -49,24 +50,40 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
         if (parent == null) {
             return;
         }
+
         final EarMetaData earConfig = parent.getAttachment(Attachments.EAR_METADATA);
```

Comment:
I think we can just return if earConfig is null here, to avoid repeatedly checkig `earConfig` later.